agent/lib/kontena/workers/log_worker.rb

```diff
@@ -51,31 +65,53 @@ def queue_processing?
 
     def start
       sleep 1 until Actor[:etcd_launcher].running?
+      start_container_workers
+      async.process_queue
```

Comment:
This still gets called twice:

* `initialize` `@queue_processing = nil` -> `async.start` -> `async.process_queue` -> `defer { while queue_processing? ... }`
* `on_connect` `@queue_processing = true`  -> `async.start` -> `async.process_queue` -> `defer { while queue_processing? ... }`

What's stopping these two `process_queue` `defer { ... }` tasks from racing and having both of them run concurrently?